## Non-diffusion or Non-protein papers
- Equivariant diffusion for molecule generation in 3d:https://arxiv.org/abs/2203.17003
- A geometric diffusion model for molecular conformation generation: https://arxiv.org/abs/2203.02923
- 3D Shape Generation and Completion through Point-Voxel Diffusion: https://arxiv.org/pdf/2104.03670.pdf
- Generative modeling of immunoglobulin proteins by direct 3d coordinate generation: https://www.biorxiv.org/content/10.1101/2020.08.07.242347v1
- While DGMs have been applied to other problems in molecular machine learning [Xu et al., 2021;
Jing et al., 2022; Hoogeboom et al., 2022],
## New Antibody generation
- [x] AbDiffuser: Full-Atom Generation of In-Vitro Functioning Antibodies: https://arxiv.org/pdf/2308.05027.pdf
- [x] Antigen-Specific Antibody Design and Optimization with Diffusion-Based Generative Models for Protein Structures: https://proceedings.neurips.cc/paper_files/paper/2022/file/3fa7d76a0dc1179f1e98d1bc62403756-Paper-Conference.pdf
- AntiBARTy Diffusion for Property Guided Antibody Design: https://arxiv.org/abs/2309.13129
- PF-ABGen: A Reliable and Efficient Antibody Generator via Poisson Flow (<span style="color:#ff0000">Maybe not a true diffusion model</span>): https://openreview.net/forum?id=fbiGerKXM6Q
## Motif design
- [x] Diffusion probabilistic modeling of protein backbones in 3d for the motif-scaffolding problem  (<span style="color:#ff0000">sample backbone -> generate sequence -> predict structure</span>): https://arxiv.org/abs/2206.04119
## DNA binding protein
 - Designing Novel DNA-Binding Proteins with Generative Deep Learning (<span style="color:#ff0000">Master thesis</span>): https://dspace.mit.edu/handle/1721.1/152003
## New protein generation
- De novo design of protein structure and function with RFdiffusion: https://www.nature.com/articles/s41586-023-06415-8
- Score-based Enhanced Sampling for Protein Molecular Dynamics: https://arxiv.org/pdf/2306.03117.pdf
- A Latent Diffusion Model for Protein Structure Generation: https://arxiv.org/abs/2305.04120
- Score-based generative modeling for de novo protein design: https://www.nature.com/articles/s43588-023-00440-3
- [x] Generative design of de novo proteins based on secondary-structure constraints using an attention-based diffusion model: https://www.cell.com/chem/pdf/S2451-9294(23)00139-0.pdf
- [x] Protein Structure and Sequence Generation with Equivariant Denoising Diffusion Probabilistic Models: https://arxiv.org/abs/2205.15019
- Protein Design with Guided Discrete Diffusion (<span style="color:#ff0000">Just sequence I think?</span>): https://arxiv.org/pdf/2305.20009.pdf
- Illuminating protein space with a programmable generative model: https://www.biorxiv.org/content/10.1101/2022.12.01.518682v1.full.pdf
- Conditional Protein Denoising Diffusion Generates Programmable Endonucleases:https://www.biorxiv.org/content/10.1101/2023.08.10.552783v1.abstract
- Diffusion probabilistic modeling of protein backbones in 3d for the motif-scaffolding problem (<span style="color:#ff0000">sample backbone -> generate sequence -> predict structure</span>): https://arxiv.org/abs/2206.04119
- Protein structure generation via folding diffusion (<span style="color:#ff0000">just structure I think?</span>): https://openreview.net/forum?id=Nkd7AS2USRd

## End-to-end/co-design
- Sidney Lyayuga Lisanza, Jacob Merle Gershon, SamWayne Kenmore Tipps, Lucas Arnoldt, Samuel
Hendel, Jeremiah Nelson Sims, Xinting Li, and David Baker. Joint generation of protein sequence
and structure with rosettafold sequence space diffusion. bioRxiv, pp. 2023–05, 2023.
- Ria Vinod, Kevin K Yang, and Lorin Crawford. Joint protein sequence-structure co-design via
equivariant diffusion. In NeurIPS 2022 Workshop on Learning Meaningful Representations of
Life, 2022.
- AbDiffuser: Full-Atom Generation of In-Vitro Functioning Antibodies: https://arxiv.org/pdf/2308.05027.pdf

## Protein-Protein Docking
- [x] DiffDock-PP: Rigid Protein-Protein Docking with Diffusion Models: https://arxiv.org/abs/2304.03889
## Sequence-to-Structure
- [x] EigenFold: Generative Protein Structure Prediction with Diffusion Models: https://arxiv.org/abs/2304.02198
- [x] State-specific protein-ligand complex structure prediction with a multi-scale deep generative model: https://arxiv.org/abs/2209.15171
- [x] End-to-end protein–ligand complex structure generation with diffusion-based generative models: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-023-05354-5
- Schneuing A, Du Y, Harris C, Jamasb A, Igashov I, Du W, Blundell T, Lió P, Gomes C, Welling M, Bronstein M, Correia B. Structure-based drug design with equivariant diffusion models. arXiv:2210.13695. 2022. https://doi.org/10.48550/ARXIV.2210.13695.
## Protein-Ligand modelling 
- [x] State-specific protein-ligand complex structure prediction with a multi-scale deep generative model: https://arxiv.org/abs/2209.15171
- [x] End-to-end protein–ligand complex structure generation with diffusion-based generative models: https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-023-05354-5
- [x] EigenFold: Generative Protein Structure Prediction with Diffusion Models: https://arxiv.org/abs/2304.02198
## Structure-to-Sequence/Inverse Folding
- Graph Denoising Diffusion for Inverse Protein Folding: https://arxiv.org/abs/2306.16819
## Protein Side-Chain Packing
- DiffPack: A Torsional Diffusion Model for Autoregressive Protein Side-Chain Packing: https://arxiv.org/abs/2306.01794
## Surveys
- A Survey on Graph Diffusion Models: Generative AI in Science for Molecule, Protein and Material (<span style="color:#ff0000">done</span>): https://arxiv.org/pdf/2304.01565.pdf
- Deep Learning Tools for Protein Binder Design: https://www.proquest.com/openview/93e77e1b7e8008d4b0522021455f59c6/1?cbl=18750&diss=y&pq-origsite=gscholar&parentSessionId=tjq89E9PY5Us1L9uwhkPsgpOnwsrHmVygpdB%2FNKig0Q%3D
- Diffusion Models in Bioinformatics: A New Wave of Deep Learning Revolution in Action: https://arxiv.org/abs/2302.10907
- Computational and artificial intelligence-based methods for antibody development: https://www.cell.com/trends/pharmacological-sciences/fulltext/S0165-6147(22)00279-6
